mlflow/langchain/__init__.py

```diff
@@ -398,7 +436,17 @@ def _load_model(path, agent_path=None, tools_path=None, agent_primitive_path=Non
     if agent_path is None and tools_path is None:
         from langchain.chains.loading import load_chain
 
-        model = load_chain(path)
+        try:
+            model = load_chain(path)
+        except ValueError as e:
+            if "`retriever` must be present." in str(e):
```

Comment:
Is there a more robust check we can do? LangChain makes breaking changes super super often, so I'm pretty worried that this error message will change or a different error message will be propagated later on

Reply:
LangChain does not have an API for checking the chain type without loading the chain. We may:
1. Implement checking the chain type in mlflow by looking at the model yaml file.
2. Contribute to LangChain to expose the API for checking the chain type without loading the chain.

2 seems more robust

Reply:
It turns out that mlflow is able to tell the model type before loading the model. RetrievalQA chains have the entry loader_fn in the MLmodel file.